### PR TITLE
Improve packaging and export of the project

### PR DESCRIPTION
**Pull request by jcarpent**
@wxmerkt I have implemented the full export of the project. It should work now. Could you check that the export is working well?

Before merging and making a new release, I will implement a small unit test to check the export is working well too.

### CODE REVIEW

**Comment by wxmerkt** [comment]
Thank you for the great effort. I will check it either tomorrow or Monday (and also check that it doesn't break MoveIt)

**Comment by jcarpent** [comment]
@wxmerkt I've added a unit test of using the new export. It seems to work well.

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,7 @@ SET(PROJECT_NAME eigenpy)
 SET(PROJECT_DESCRIPTION "Bindings between Numpy and Eigen using Boost.Python")
 SET(PROJECT_URL "http://github.com/stack-of-tasks/eigenpy")
 SET(PROJECT_USE_CMAKE_EXPORT TRUE)
+SET(PROJECT_USE_KEYWORD_LINK_LIBRARIES TRUE)
```
Comment (wxmerkt) [inline CMakeLists.txt]: What does this do?

Comment (wxmerkt) [reply CMakeLists.txt]: Nevermind, just found it.

**Comment by jcarpent** [comment]
@wxmerkt A kind reminder. If you have time today to check this PR ;)

**Comment by wxmerkt** [comment]
I have just tested it and it appears to work.

However, please note: The install path for the `eigenpyConfig.cmake` is changing from `share/cmake/eigenpy` to `lib/cmake/eigenpy` (!!).

**Comment by jcarpent** [comment]
> The install path for the `eigenpyConfig.cmake` is changing from `share/cmake/eigenpy` to `lib/cmake/eigenpy` (!!).

Is it an issue?

**Comment by wxmerkt** [comment]
It appears to work - we should note it in case someone comes back in future if it breaks something. From me, this PR is good to go

**Comment by jcarpent** [comment]
Do you want me to install it in the share folder too?

**Comment by wxmerkt** [comment]
No, let's leave it as is (no duplication) until we have a use case that requires it

**Comment by jcarpent** [comment]
@wxmerkt Thanks a lot for your nice feedback.

**Pull request merged**